code/go/0chain.net/smartcontract/dbs/types.go

```diff
@@ -0,0 +1,82 @@
+package dbs
+
+type Provider int
+
+const (
+	Miner Provider = iota
+	Sharder
+	Blobber
+	Validator
+	Authorizer
+)
+
+type PoolStatus int
```

Comment:
this is redundant too. it'll become difficult to maintain this redundancy
maybe we should think about constants/stakepool kindof package here